### PR TITLE
Refactor of eager/lazy loading

### PR DESCRIPTION
**Pull request by avree**
1. Template literals for error messages. 
2. loadingMode duplicate code removal. 
3. better _return variable name. 
4. Lazy/eager loading split into functions.

### CODE REVIEW

**Review by reatlat — CHANGES_REQUESTED** [review]
@avree I really like how you think, but could we keep human readable code as a comment? so we can back in future and manage it?

**Comment by avree** [comment]
> @avree I really like how you think, but could we keep human readable code as a comment? so we can back in future and manage it?

Yes, that makes sense to me. I'll add back the comments when I'm back to my machine, unless you want to add them sooner.

File: .eleventy.js [hunk 1]
```diff
@@ -50,122 +48,48 @@ module.exports = (eleventyConfig, options = {}) => {
         return pattern.test(str);
     };
 
+    function lazyLoadingCode(encodedConfig, options, hsScripts) {
```
Comment (reatlat) [inline .eleventy.js:51]: @avree this is just a personal preferences, but. I think declare functions as constant above/before executing is better way, correct me if I wrong :)

Comment (reatlat) [reply .eleventy.js:51]: or maybe transform URL validator into a function declaration, instead of a constant... so make it consistence 🧐

**Review by reatlat — APPROVED** [review]
LGTM, @avree Thanks

**Pull request merged**